mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/SolveMessageInconsistenciesTask.java

```diff
@@ -0,0 +1,136 @@
+
+    public static class Details implements TaskExecutionDetails.AdditionalInformation {
+        private final Instant instant;
+        private final long processedImapUidEntries;
+        private final long processedMessageIdEntries;
+        private final long addedMessageIdEntries;
+        private final long updatedMessageIdEntries;
+        private final long removedMessageIdEntries;
+        private final ImmutableList<String> fixedInconsistencies;
+        private final ImmutableList<String> errors;
+
+        public Details(Instant instant, long processedImapUidEntries, long processedMessageIdEntries,
+                       long addedMessageIdEntries, long updatedMessageIdEntries, long removedMessageIdEntries,
+                       ImmutableList<String> fixedInconsistencies, ImmutableList<String> errors) {
+            this.instant = instant;
+            this.processedImapUidEntries = processedImapUidEntries;
+            this.processedMessageIdEntries = processedMessageIdEntries;
+            this.addedMessageIdEntries = addedMessageIdEntries;
+            this.updatedMessageIdEntries = updatedMessageIdEntries;
+            this.removedMessageIdEntries = removedMessageIdEntries;
+            this.fixedInconsistencies = fixedInconsistencies;
+            this.errors = errors;
+        }
+
+        @Override
+        public Instant timestamp() {
+            return instant;
+        }
+
+        @JsonProperty("processedImapUidEntries")
+        public long getProcessedImapUidEntries() {
+            return processedImapUidEntries;
+        }
+
+        @JsonProperty("processedMessageIdEntries")
+        public long getProcessedMessageIdEntries() {
+            return processedMessageIdEntries;
+        }
+
+        @JsonProperty("addedMessageIdEntries")
+        public long getAddedMessageIdEntries() {
+            return addedMessageIdEntries;
+        }
+
+        @JsonProperty("updatedMessageIdEntries")
+        public long getUpdatedMessageIdEntries() {
+            return updatedMessageIdEntries;
+        }
+
+        @JsonProperty("removedMessageIdEntries")
+        public long getRemovedMessageIdEntries() {
+            return removedMessageIdEntries;
+        }
+
+        @JsonProperty("fixedInconsistencies")
+        public ImmutableList<String> getFixedInconsistencies() {
+            return fixedInconsistencies;
+        }
+
+        @JsonProperty("errors")
+        public ImmutableList<String> getErrors() {
+            return errors;
+        }
+    }
+
+    private final SolveMessageInconsistenciesService service;
+    private Context context;
+
+    public SolveMessageInconsistenciesTask(SolveMessageInconsistenciesService service) {
+        this.service = service;
+        this.context = new Context();
+    }
+
+    @Override
+    public Result run() {
+        return service.fixMessageInconsistencies(context)
+            .block();
```

Comment:
Should maybe subscribe on the scheduler, like other tasks?

```
return service.fixMessageInconsistencies(context)
            .subscribeOn(Schedulers.elastic())
            .block();
```

Reply:
Not sure, the caller of `run` is supposed to take care of running the task in a dedicated thread, don't you think?